File: src/ios/MfNiWrapperPlugin.m
#import <NICardManagementSDK-Swift.h>
#import <Cordova/CDV.h>
#import "MfNiWrapperPlugin.swift"

@interface MfNiWrapperPlugin : CDVPlugin {
    // Member variables go here.
}
@end

@implementation MfNiWrapperPlugin
    
    // Plugin initialization
    - (void)pluginInitialize {
        NSLog(@"MfNiWrapperPlugin - pluginInitialize");
    }

    // - (void) viewCardDetails:(CDVInvokedUrlCommand*)command
    // {
    //     NSLog(@"MfNiWrapperPlugin - viewCardDetails");
    //     [self.commandDelegate runInBackground:^{
    //         // NSString* payload = [command.arguments objectAtIndex:0];
    //         // NSString* callbackId = command.callbackId;
    //         // NSString* msg = [NSString stringWithFormat: @"MfNiWrapperPlugin - viewCardDetails - payload: %@", payload];
    //         // NSLog(@"%@", msg);
    //         // CDVPluginResult* result = [CDVPluginResult
    //         //                            resultWithStatus:CDVCommandStatus_OK
    //         //                            messageAsString:msg];
    //         // [self.commandDelegate sendPluginResult:result callbackId:callbackId];
    //         [self.viewCardDetails];
    //     }];
    // }
@end

File: src/ios/MfNiWrapperPlugin.swift
import UIKit
import NICardManagementSDK
import Combine

// let sdk = NICardManagementAPI(
//             rootUrl: rootUrl,
//             cardIdentifierId: cardIdentifierId,
//             cardIdentifierType: cardIdentifierType,
//             bankCode: bankCode,
//             tokenFetchable: NICardManagementTokenFetchable
// )

@objc(MfNiWrapperPlugin) class MfNiWrapperPlugin : CDVPlugin {
    
    // Your other properties and methods
    
    @objc func echo(_ command: CDVInvokedUrlCommand) {
        let message = command.arguments[0] as? String ?? ""
        print("Echo: \(message)")
        
        let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: "Echoed: \(message)")
        self.commandDelegate!.send(pluginResult, callbackId: command.callbackId)
    }
}

